src/fmgc/src/efis/EfisVectors.ts

```diff
@@ -0,0 +1,177 @@
+import { GuidanceController } from '@fmgc/guidance/GuidanceController';
+import { EfisSide, EfisVectorsGroup, Mode } from '@shared/NavigationDisplay';
+import { ArcPathVector, LinePathVector, PathVector } from '@fmgc/guidance/lnav/PathVector';
+import { LnavConfig } from '@fmgc/guidance/LnavConfig';
+import { LateralMode } from '@shared/autopilot';
+import { withinEditArea } from '@fmgc/efis/EfisCommon';
+import { TaskCategory } from '@fmgc/guidance/TaskQueue';
+
+const TRANSMIT_GROUP_SIZE = 4;
+
+const UPDATE_TIMER = 2_500;
+
+export class EfisVectors {
+    private listener = RegisterViewListener('JS_LISTENER_SIMVARS');
+
+    constructor(
+        private guidanceController: GuidanceController,
+    ) {
+    }
+
+    private currentActiveVectors = [];
+
+    private currentDashedVectors = [];
+
+    private currentTemporaryVectors = [];
+
+    public forceUpdate() {
+        this.updateTimer = UPDATE_TIMER * 0.8;
+    }
+
+    private updateTimer = 0;
+
+    public update(deltaTime: number): void {
+        this.updateTimer += deltaTime;
+
+        if (this.updateTimer < UPDATE_TIMER) {
+            return;
+        }
+
+        this.updateTimer = 0;
+
+        if (LnavConfig.DEBUG_PERF) {
+            console.time('vectors transmit');
+        }
+
+        const activeFlightPlanVectors = this.guidanceController.activeGeometry?.getAllPathVectors() ?? [];
+        const temporaryFlightPlanVectors = this.guidanceController.temporaryGeometry?.getAllPathVectors() ?? [];
+
+        const visibleActiveFlightPlanVectors = activeFlightPlanVectors
+            .filter((vector) => this.vectorWithinCurrentEditArea(vector, 'L') || this.vectorWithinCurrentEditArea(vector, 'R'));
+        const visibleTemporaryFlightPlanVectors = temporaryFlightPlanVectors
+            .filter((vector) => this.vectorWithinCurrentEditArea(vector, 'L') || this.vectorWithinCurrentEditArea(vector, 'R'));
+
+        // ACTIVE
+
+        const engagedLateralMode = SimVar.GetSimVarValue('L:A32NX_FMA_LATERAL_MODE', 'Number') as LateralMode;
+        const armedLateralMode = SimVar.GetSimVarValue('L:A32NX_FMA_LATERAL_ARMED', 'Enum');
+        const navArmed = (armedLateralMode >> 0) & 1;
+
+        const transmitActive = engagedLateralMode === LateralMode.NAV || navArmed;
```

Comment:
Active vectors should be transmitted if any lateral autocontrol mode is active, so if either NAV or LOC is armed, or if NAV, LOC, LOC*, LAND, FLARE, ROLL OUT, RWY, GA TRK or RWY TRK is active
```suggestion
        const navArmed = (armedLateralMode >> 0) & 1;
        const navActive = engagedLateralMode === LateralMode.NAV;
        const takeoffGoaroundModeActive = engagedLateralMode === LateralMode.RWY || engagedLateralMode === LateralMode.RWY_TRACK || engagedLateralMode === LateralMode.GA_TRACK;
        const locArmed = (armedLateralMode >> 1) & 1;
        const locActive = engagedLateralMode >= LateralMode.LOC_CPT && engagedLateralMode <= LateralMode.ROLL_OUT;

        const transmitActive = navActive || takeoffGoaroundModeActive || locActive || navArmed || locArmed;
```